crates/uv-git/src/git.rs

```diff
@@ -969,13 +969,34 @@ pub(crate) fn fetch(
 
     debug!("Performing a Git fetch for: {remote_url}");
     match strategy {
-        FetchStrategy::Cli => fetch_with_cli(repo, remote_url, &refspecs, tags),
+        FetchStrategy::Cli => {
+            // Try each refspec
+            let results = refspecs
+                .into_iter()
+                .map(|refspec| fetch_with_cli(repo, remote_url, &[refspec], tags))
+                .collect::<Vec<_>>();
+
+            // Only fail if _all_ refspecs cannot be found
+            // This matches the Libgit2 behavior
+            if results.iter().all(Result::is_err) {
```

Comment:
Small performance nit: `if !results.iter().any(Result::is_ok)` would be slightly faster as a check since it would return early.

Reply:
Won't `all` exit early on the first `Result::Ok` too?

Reply:
Right, sorry. My bad 🤦‍♀️